braindecode/augmentation/transforms.py

```diff
@@ -895,4 +890,17 @@ def __init__(
         )
 
     def get_params(self, X, y):
-        return self.alpha, self.beta_per_sample, self.rng
+        device = X.device
+        batch_size, _, _ = X.shape
+
+        if self.alpha > 0:
+            if self.beta_per_sample:
+                lam = torch.as_tensor(
+                    self.rng.beta(self.alpha, self.alpha, batch_size)
+                ).to(device)
+            else:
+                lam = torch.ones(batch_size).to(device)
+                lam *= self.rng.beta(self.alpha, self.alpha)
+        else:
+            lam = torch.ones(batch_size).to(device)
+        return lam, self.rng
```

Comment:
Great!

If we don't want to pass the 'rng', we could also pass the permutated index to the functional.

Reply:
Indeed. It would even make more sense to pass (X1, y1) and (X2, y2) wouldn't it? So that the functional does something really atomic and fully deterministic. The transform could handle the sampling of the permuted index and use it to get (X2, y2) from (X1, y1). What do you think ?

Reply:
Doesn't `functional` always gets `X` and `y`? We would need to pass `lam` anyway, so `X, y, lam, idx_perm` would be cleaner, I guess.

Reply:
I'm ok with that too :) Go for it